mteb/abstasks/MTEBResults.py

```diff
@@ -138,9 +142,94 @@ def to_disk(self, path: Path) -> None:
             f.write(self.model_dump_json())
 
     @classmethod
-    def from_disk(cls, path: Path) -> MTEBResults:
+    def from_disk(cls, path: Path, load_historic_data: bool = False) -> MTEBResults:
+        """Load MTEBResults from disk.
+
+        Args:
+            path: The path to the file to load.
+            load_historic_data: Whether to attempt to load historic data from before v1.11.0.
+        """
         with path.open("r") as f:
-            return cls.model_validate(json.load(f))
+            data = json.load(f)
+
+        if not load_historic_data:
+            try:
+                return cls.model_validate(data)
+            except Exception as e:
+                raise ValueError(
+                    f"Error loading MTEBResults from disk. You can try to load historic data by setting `load_historic_data=True`. Error: {e}"
+                )
+
+        err: str = ""
+        try:
+            return cls.model_validate(data)
+        except Exception as e:
+            err = f"Error loading MTEBResults from disk: {e}"
+        try:
+            if "mteb_version" in data and Version(data["mteb_version"]) < Version(
+                "1.11.0"
+            ):
+                logger.info(
+                    f"Could not load MTEBResults from disk, got error: {err}. Attempting to load from disk using format from before v1.10.0"
+                )
+                return cls._convert_from_before_v1_11_0(data)
+        except Exception as historic_err:
+            err += f"\nAttempted to load as historic data but got the error: {historic_err}"
+        raise ValueError(err)
+
+    @classmethod
+    def _convert_from_before_v1_11_0(cls, data: dict) -> MTEBResults:
+        from mteb.overview import TASKS_REGISTRY
+
+        scores = {**data}
+
+        dataset_revision = scores.pop("dataset_revision")
+        task_name = scores.pop("mteb_dataset_name")
+        mteb_version = scores.pop("mteb_version")
+
+        # calculate evaluation time across all splits (move to top level)
+        evaluation_time = 0
+        for split, split_score in scores.items():
+            if "evaluation_time" in split_score:
+                evaluation_time += split_score.pop("evaluation_time")
```

Comment:
Why are we doing this? Isn't moving this to the top level a loss of information? 🤔 Maybe having a `total_evaluation_time` at the top level or sth if it is important to have?

Reply:
It is. It is a decision made in the PR #759, I believe it should either be at the top-level or at the hf_subset level (so one level deeper than what it is now), however, that requires a larger refactor. During it this way we can always add it later.

having it somewhere in between creates an inconsistent structure which will require quite a few exceptions and would lead to errors is you have a huggingface subset called "evaluation_time" (which would be odd)

Reply:
Okay I see; I guess top level only is fine then

Reply:
I would love to add the subset level as well though (totally agree with your point that it is a loss of information)